src/zcash/address/zip32.cpp

```diff
@@ -19,12 +19,26 @@ const unsigned char ZCASH_HD_SEED_FP_PERSONAL[BLAKE2bPersonalBytes] =
 const unsigned char ZCASH_TADDR_OVK_PERSONAL[BLAKE2bPersonalBytes] =
     {'Z', 'c', 'T', 'a', 'd', 'd', 'r', 'T', 'o', 'S', 'a', 'p', 'l', 'i', 'n', 'g'};
 
-HDSeed HDSeed::Random(size_t len)
+const libzcash::diversifier_index_t MAX_TRANSPARENT_CHILD_IDX(0x40000000);
+
+MnemonicSeed MnemonicSeed::Random(uint32_t bip44CoinType, Language language, size_t entropyLen)
 {
-    assert(len >= 32);
-    RawHDSeed rawSeed(len, 0);
-    GetRandBytes(rawSeed.data(), len);
-    return HDSeed(rawSeed);
+    assert(entropyLen >= 32);
+    while (true) {
+        std::vector<unsigned char> entropy(entropyLen, 0);
+        GetRandBytes(entropy.data(), entropyLen);
+        const char* phrase = zip339_entropy_to_phrase(language, entropy.data(), entropyLen);
+        std::string mnemonic(phrase);
+        zip339_free_phrase(phrase);
+        MnemonicSeed seed(language, mnemonic);
+
+        // Verify that the seed data is valid entropy for unified spending keys at
+        // account 0 and account 0x7FFFFFFE
+        if (libzcash::UnifiedSpendingKey::Derive(seed, bip44CoinType, 0).has_value() &&
+            libzcash::DeriveZip32TransparentSpendingKey(seed, bip44CoinType, ZCASH_LEGACY_TRANSPARENT_ACCOUNT).has_value())  {
```

Comment:
```suggestion
        if (libzcash::UnifiedSpendingKey::Derive(seed, slip44CoinType, 0).has_value() &&
            libzcash::DeriveZip32TransparentSpendingKey(seed, slip44CoinType, ZCASH_LEGACY_TRANSPARENT_ACCOUNT).has_value())  {
```

Reply:
This is the bip44 coin type in terms of the position that it occupies in the derivation path.